### PR TITLE
Jenkins file

### PR DESCRIPTION
**Pull request by fsufitch**

### CODE REVIEW

File: ci/archive.sh [hunk 1]
```diff
@@ -8,4 +8,4 @@ source $root/ci/vars.sh
 
 cp nginx.conf dist
 
-tar -czf $APP.$EXT -C $root dist
+zip $APP.$EXT $root dist
```
Comment (go0ty) [inline ci/archive.sh]: Curious - why change from tarball to zip?

File: JenkinsFile [hunk 2]
```diff
@@ -1,18 +1,222 @@
...
+    }
+    withEnv(['CF_HOME=.cf']) {
+      withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: "${env.PCF_CREDS}", usernameVariable: 'CFUSER', passwordVariable: 'CFPASS']]) {
+        sh "cf api ${env.PCF_API_ENDPOINT}"
+        sh "cf auth ${CFUSER} ${CFPASS}"
+        sh "cf target -o ${env.PCF_ORG} -s ${env.PHASE_ONE_PCF_SPACE}"
+        sh "cf push ${appName} -f manifest.jenkins.yml --hostname ${appName} -d ${env.PHASE_ONE_PCF_DOMAIN} --no-start"
+        sh "cf set-env ${appName} SPACE ${env.PHASE_ONE_PCF_SPACE}"
+        sh "cf set-env ${appName} DOMAIN ${env.PHASE_ONE_PCF_DOMAIN}"
+        sh "cf start ${appName}"
```
Comment (go0ty) [inline JenkinsFile]: We need to handle failures by deleting the app, or else we end up with hanging apps that never get deleted.

https://github.com/venicegeo/pz-gateway/blob/master/JenkinsFile#L108

File: JenkinsFile [hunk 3]
```diff
@@ -1,18 +1,222 @@
...
+      }
+      withEnv(['CF_HOME=.cf']) {
+        withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: "${env.PCF_CREDS}", usernameVariable: 'CFUSER', passwordVariable: 'CFPASS']]) {
+          sh "cf api ${env.PCF_API_ENDPOINT}"
+          sh "cf auth ${CFUSER} ${CFPASS}"
+          sh "cf target -o ${env.PCF_ORG} -s ${env.PHASE_TWO_PCF_SPACE}"
+          sh "cf push ${appName} -f manifest.jenkins.yml --hostname ${appName} -d ${env.PHASE_TWO_PCF_DOMAIN} --no-start"
+          sh "cf set-env ${appName} SPACE ${env.PHASE_TWO_PCF_SPACE}"
+          sh "cf set-env ${appName} DOMAIN ${env.PHASE_TWO_PCF_DOMAIN}"
+          sh "cf start ${appName}"
```
Comment (go0ty) [inline JenkinsFile:152]: https://github.com/venicegeo/pz-gateway/blob/master/JenkinsFile#L108

File: ci/archive.sh (hunk 1 above)
Comment (fsufitch) [reply ci/archive.sh]: The `mvn dependency:get` command has `-Dpackaging=zip` set, and I cannot find documentation for what the string is for setting it to tgz (and omitting it defaults to jar, which is not right).

**Review by go0ty — APPROVED** [review]

**Pull request merged**